forge/lang/expander.rkt

```diff
@@ -724,6 +748,38 @@
        (syntax/loc stx (begin block.test-decls ...))
        (syntax/loc stx (begin)))]))
 
+
+; PropertyWhereDecl : PROPERTY-TOK Name OF-TOK Name Expr WHERE-TOK TEST-CONSTRUCT*
+(define-syntax (PropertyWhereDecl stx)
+  (syntax-parse stx
+  [((~literal PropertyWhereDecl) 
+                              (~and constraint-type (~or "overconstraint" "underconstraint"))  
+                              prop-name:NameClass
+                              "of"
+                              pred-name:NameClass
+                              prop-expr:BlockClass 
+                              (~optional scope:ScopeClass)
+                              (~optional bounds:BoundsClass)
+                               "where"
+                              where-blocks:TestConstructClass ...  ) 
+
+  (with-syntax ([test_name (format-id stx "~a-subproperty" (syntax/loc stx prop-name.name) #:source stx)]
+                ;;; Overconstraint : Prop => Pred
+                ;;; Underconstraint Pred => Prop
+                [imp_total (syntax-e (if (equal? (syntax-e #'constraint-type) "overconstraint") 
```

Comment:
Delete this `syntax-e`.

If that ends up changing important source locations, then use `syntax/loc` instead of `syntax` below.